02-getting_started.Rmd

```diff
@@ -1,67 +1,212 @@
+Bug fixes and new features are introduced in `r-devel` first. 
+If the change meets the [development guidelines](https://developer.r-project.org/devel-guidelines.txt) R Core will also make the change in `r-patched`.
 
-3. Select the language while installing, read the GNU general public license information, and select the destination location to start the installation. You will be prompted to select components at this stage: `User installation`, `32-bit User installation`, `64-bit User installation`, or `Custom installation`. The default option may be chosen for the questions from this step onwards to complete the installation. 
+## Prerequisites  {.tabset}
 
-## Building R and R packages
+To install from the source code you will need the source code and the dependencies of R.
 
-### What tools do you need to build R from source on Windows?
+Retrieve R source code via into `TOP_SRCDIR`, note that we retrieve the r-devel source code:
 
-1. [RTools](https://github.com/r-windows/docs/blob/master/faq.md#what-is-rtools) is the toolchain bundle that you can use to build R base and R packages containing compiled code, on Windows.
+      export TOP_SRCDIR="$HOME/Downloads/R"
+      svn checkout https://svn.r-project.org/R/trunk/ $TOP_SRCDIR
 
-2. You also need a distribution of LaTeX installed for building R and checking packages. The MiKTeX distribution of LaTeX that is used on CRAN can be downloaded from https://miktex.org.
+If you need to install svn you can use your distribution's package manager to install it.
 
-### How to setup `RTools`?
+### Ubuntu
 
-1. The latest version of `RTools` can be downloaded from https://cran.r-project.org/bin/windows/Rtools/ and run in the Windows-style installer. You will need to know if you have a 32-bit or 64-bit Windows machine (right-click `This PC` in Windows Explorer and check the properties if you are unsure).
+In Ubuntu you can use :
 
-2. Don't forget to add `RTools` to the path as documented on the download page.
+      sudo apt-get install -y \
+          subversion \
+          autoconf \
+          automake \
+          autopoint \
+          bash \
+          bison \
+          bzip2 \
+          flex \
+          g++ \
+          g++-multilib \
+          gettext \
+          git \
+          gperf \
+          intltool \
+          libc6-dev-i386 \
+          libgdk-pixbuf2.0-dev \
+          libltdl-dev \
+          libssl-dev \
+          libtool-bin \
+          libxml-parser-perl \
+          lzip \
+          make \
+          openssl \
+          p7zip-full \
+          patch \
+          perl \
+          python3 \
+          python3-mako \
+          python3-setuptools \
+          python \
+          ruby \
+          sed \
+          unzip \
+          wget \
+          xz-utils \
+          texinfo \
+          sqlite3 \
+          zstd
```

Comment:
Where did this list come from? Can we not just do `sudo apt-get build-dep r-base`? (plus subversion). You can see what would be installed this way via e.g. https://launchpad.net/ubuntu/kinetic/+source/r-base

Reply:
Maybe also check the output of `apt-rdepends --build-depends --follow=DEPENDS r-base` as suggested here: https://askubuntu.com/a/21394. But I guess the advantage of simply using `sudo apt-get build-dep r-base` (assuming it is sufficient) is that we do not need to keep such a list up to date in the guide.

Reply:
I will test it tonight, but I think rdepends provides the list of reverse dependencies not direct dependencies (all cran packages installed via cran2u), but I will try it and compare the output.

Reply:
@HenrikBengtsson It has not been discussed, you mentioned it in the r-devel slack some time ago that RStudio had a [list of dependencies](https://docs.rstudio.com/resources/install-r-source/#install-required-dependencies).
Ideally it would be nice to have this list for other distributions but I don't have any non-Debian machine at hand. I might ask Iñaki Úcar, who maintains the [fedora distribution of R](https://cran.r-project.org/bin/linux/fedora/), if he could help providing such a list.

apt-rdepends returns (`apt-rdepends --build-depends --follow=DEPENDS r-base | grep " B" | sed -e "s/  Build-Depends: //" | sort`):

```
bash-completion
bison
debhelper-compat (= 11)
default-jdk
g++ (>= 4:4.9.2-2)
gcc (>= 4:4.9.2-2)
gfortran (>= 4:4.9.2-2)
groff-base
libblas-dev
libbz2-dev
libcairo2-dev
libcurl4-dev
libcurl4-openssl-dev
libjpeg-dev
liblapack-dev
liblzma-dev
libncurses5-dev
libpango1.0-dev
libpcre2-dev
libpcre3-dev
libpng-dev
libreadline-dev
libtiff5-dev
libx11-dev
libxt-dev
mpack
tcl8.6-dev
texinfo (>= 4.1-2)
texlive-base
texlive-extra-utils
texlive-fonts-extra
texlive-fonts-recommended
texlive-latex-base
texlive-latex-extra
texlive-latex-recommended
texlive-plain-generic
tk8.6-dev
x11proto-core-dev
xauth
xdg-utils
xfonts-base
xvfb
zlib1g-dev
```
This is much shorter (43 vs 14X) and realistic imho, but perhaps users that have MiKTeX use it instead of [TeX Live](https://www.tug.org/texlive/).

Reply:
@HenrikBengtsson we should bear in mind who this guide is for: it is intended for individual contributors to get up and running, especially those with little experience building from source. That is why we focused initially on Windows, as the most common user OS (even among developers: https://survey.stackoverflow.co/2022/#operating-system) and also where users typically install binaries. System admins or advanced users that want to customize their R installation can always refer to R-admin.

So I would prefer to keep it simple if possible, perhaps with a separate section on advanced customization that could be added later. I am quite happy to rely on a third party for maintaining the list as they are the ones expert in the respective distribution.

It would be good to cover Fedora based distros too though. From the link that @llrs shared, it seems that `sudo yum-builddep R` works after enabling the EPEL repository? We may have to separate this out into a new issue if we need to find a collaborator with a Fedora-based distro to test. In the meantime we can make it clear that instructions are for Debian-based distros.